lib/puppet/module.rb

```diff
@@ -33,6 +33,25 @@ def self.find(modname, environment = nil)
     env.module(modname)
   end
 
+  def self.is_module_directory?(name, path)
+    # it must be a directory
+    fullpath = File.join(path, name)
+    return false unless Puppet::FileSystem.directory?(fullpath)
+    return is_module_directory_name?(name)
+  end
+
+  def self.is_module_directory_name?(name)
+    # it must match an installed module name according to forge validator
+    return true if name =~ /^[a-z][a-z0-9_]*$/
+    return false
+  end
+
+  def self.is_module_namespaced_name?(name)
+    # it must match the full module name according to forge validator
+    return true if name =~ /^[a-zA-Z0-9]+[-][a-z][a-z0-9_]*$/   
+    return false
```

Comment:
@adreyer pointed out =~ doesn't actually return a bool. And I found http://stackoverflow.com/questions/2215303/ruby-whats-the-proper-syntax-for-a-boolean-regex-method

But tldr, I'm okay with this as is.